.github/workflows/GenerateAsyncCode.yml

```diff
@@ -0,0 +1,42 @@
+name: Generate Async code
+
+on: 
+  pull_request_target:
+    paths:
+      - '**.cs'
+
+permissions: {}
+
+jobs:
+  generate-async:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+        repository: ${{ github.event.pull_request.head.repo.full_name }}
+        token: ${{ secrets.NHIBERNATE_BOT_TOKEN }}
```

Comment:
I have tried with GITHUB_TOKEN but it seems there is no way to give it right to push a commit to the PR. I am not even sure the trouble is here.

My tests just fails on my fork. fredericDelaporte/NHibernate-Caches#2.

If I have to use that NHIBERNATE_BOT_TOKEN [secret](https://github.com/nhibernate/NHibernate-Caches/settings/secrets/actions), it would have to be created, and what should it be? A PAT from the [nhibernate-bot account](https://github.com/nhibernate-bot)? (I have no access to it.)

Reply:
> A PAT from the nhibernate-bot account


Yes

Reply:
Ok, how can I get one? Or is-it something you have to do yourself?

Reply:
I've added token to the organization.

Reply:
Thanks.